jenkins/ci.suse.de/pipelines/openstack-ardana-gating.Jenkinsfile

```diff
@@ -0,0 +1,116 @@
+/**
+ * The openstack-ardana-gating Jenkins Pipeline
+ */
+
+def ardana_lib = null
+
+pipeline {
+  // skip the default checkout, because we want to use a custom path
+  options {
+    skipDefaultCheckout()
+    timestamps()
+  }
+
+  agent {
+    node {
+      label 'cloud-trigger'
+      customWorkspace "${JOB_NAME}-${BUILD_NUMBER}"
+    }
+  }
+
+  stages {
+
+    stage('Setup workspace') {
+      steps {
+        script {
+          env.staging_build = sh (
+            returnStdout: true,
+            script: "wget -q -O - http://provo-clouddata.cloud.suse.de/repos/x86_64/SUSE-OpenStack-Cloud-${version}-devel-staging/media.1/build | grep -oP 'Build[0-9]+'"
+          ).trim()
+          currentBuild.displayName = "#${BUILD_NUMBER}: ${staging_build}"
+
+          env.deploy_job_name = "cloud-ardana${version}-job-std-min-x86_64"
+          env.update_job_name = "cloud-ardana${version}-job-std-3cp-devel-staging-updates-x86_64"
+
+          sh('''
+            git clone $git_automation_repo --branch $git_automation_branch automation-git
+          ''')
+
+          env.starttime = sh (
+            returnStdout: true,
+            script: 'echo $(date +%s)'
+          ).trim()
+
+          ardana_lib = load "automation-git/jenkins/ci.suse.de/pipelines/openstack-ardana.groovy"
+        }
+      }
+    }
+
+    stage('Run validation jobs') {
+      // abort all stages if one of them fails
+      failFast true
+      parallel {
+
+        stage('Run deploy job') {
+          steps {
+            script {
+              // reserve a resource here for the openstack-ardana job, to avoid
+              // keeping a cloud-ardana-ci worker busy while waiting for a
+              // resource to become available.
+              lock(label: ardana_env,
+                   variable: 'reserved_env',
+                   quantity: 1) {
+                ardana_lib.trigger_build("${deploy_job_name}", [
+                  string(name: 'ardana_env', value: reserved_env),
+                  string(name: 'reserve_env', value: "false"),
+                  string(name: 'git_automation_repo', value: "$git_automation_repo"),
+                  string(name: 'git_automation_branch', value: "$git_automation_branch")
+                ], false)
+              }
+            }
+          }
+        }
+
+        stage('Run deploy-and-update job') {
```

Comment:
We should only run this on Cloud 8. Cloud 9 is still in development so the update can fail.

Reply:
Done